File: 0060-permutation-sequence/0060-permutation-sequence.js
/**
 * @param {number} n
 * @param {number} k
 * @return {string}
 */
var getPermutation = function(n, k) {
    let fact = 1;
    const numbers = [];
    
    for(let i = 1; i < n; i++){
        fact = fact * i;
        numbers.push(i);
    }

    numbers.push(n);

    let ans = "";
    k = k - 1;

    while(true){
        const index = Math.floor(k / fact);
        ans = ans + numbers[index];
        numbers.splice(index, 1); // Remove used number

        if(numbers.length === 0) break;

        k = k % fact;
        fact = Math.floor(fact / numbers.length);
    }

    return ans;
};

File: 0060-permutation-sequence/0060-permutation-sequence.py
class Solution:
    def getPermutation(self, n: int, k: int) -> str:
        numbers = list(range(1, n + 1))
        factorial = math.factorial(n - 1)
        k -= 1 
        result = ""

        for i in range(n - 1, -1, -1):
            index = k // factorial
            result += str(numbers.pop(index))
            if i > 0:
                k %= factorial
                factorial //= i

        return result


    def getPermutation1(self, n: int, k: int) -> str:

        nums = []
        result = []

        for i in range(1, n + 1):
            nums.append(i)
        
        def backtrack(ds):
            if len(ds) == n:
                result.append(ds[:])
                print(ds)

            for i in range(len(nums)):
                if nums[i] in ds:
                    continue
                
                ds.append(nums[i])
                backtrack(ds)
                ds.pop()

        backtrack([])

        arr = result[k - 1]

        ans = ""
        for num in arr:
            ans += str(num)
        return ans
        